src/Transfer/Destination.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Utopia\Transfer;
+
+abstract class Destination extends Target
+{
+    /**
+     * Source
+     *
+     * @var Source $source
+     */
+    protected Source $source;
+
+    /**
+     * Get Source
+     *
+     * @return Source
+     */
+    public function getSource(): Source
+    {
+        return $this->source;
+    }
+
+    /**
+     * Set Soruce
+     *
+     * @param Source $source
+     *
+     * @return self
+     */
+    public function setSource(Source $source): self
+    {
+        $this->source = $source;
+        return $this;
+    }
+
+    /**
+     * Transfer Resources to Destination from Source callback
+     *
+     * @param array $resources
+     * @param callable $callback
+     */
+    public function run(array $resources, callable $callback): void
+    {
+        $this->source->run($resources, function (array $resources) use ($callback) {
+            $this->importResources($resources, $callback);
+        });
+    }
+
+    /**
+     * Import Resources
+     *
+     * @param array $resources
+     * @param callable $callback (array $resources)
+     *
+     */
+    abstract public function importResources(array $resources, callable $callback): void;
```

Comment:
What's the difference between run and importResources? If needed, can we rename it to just `import`? Why is it public? Can it be private if used only by run?

Reply:
Private methods cannot be abstract however we can make it protected